gui/journal.lua

```diff
@@ -48,69 +177,184 @@ function JournalWindow:sanitizeFrame(frame)
     return frame
 end
 
-function JournalWindow:postUpdateLayout()
-    self:saveConfig()
-end
-
 function JournalWindow:saveConfig()
+    if not self.save_layout then
+        return
+    end
+
+    local toc = self.subviews.table_of_contents_panel
+
     utils.assign(journal_config.data, {
-        frame = self.frame
+        frame = self.frame,
+        toc = {
+            width = toc.frame.w,
+            visible = toc.visible
+        }
     })
     journal_config:write()
 end
 
+function JournalWindow:loadConfig()
+    if not self.save_layout then
+        return nil, false, 25
+    end
+
+    local window_frame = copyall(journal_config.data.frame or {})
+    local table_of_contents = copyall(journal_config.data.toc or {
+        width=20,
+        visible=false
+    })
+
+    return window_frame, table_of_contents.visible or false, table_of_contents.width or 25
+end
+
+function JournalWindow:onPanelResizeBegin()
+    self.resizing_panels = true
+end
+
+function JournalWindow:onPanelResizeEnd()
+    self.resizing_panels = false
+    self:ensurePanelsRelSize()
+
+    self:updateLayout()
+end
+
+function JournalWindow:onRenderBody(painter)
```

Comment:
it would be useful here to set the current ToC list selection to the currently section, or rather, the section that the 5th line from the top of the screen contains if the cursor is not onscreen, and the section that contains the cursor if the cursor is onscreen.